### PR TITLE
FIX: Quoting a quote preserves the original post information

### PR DESCRIPTION
**Pull request by nbianca**
Let's say post #2 quotes post number #1. If a user decides to quote the
quote in post #2, it should keep the information of post #1
("user_1, post: 1, topic: X"), instead of replacing with current post
info ("user_2, post: 2, topic: X").

### CODE REVIEW

File: app/assets/javascripts/discourse/components/quote-button.js.es6 [hunk 1]
```diff
@@ -48,8 +48,28 @@ export default Component.extend({
       }
     }
 
+    let opts;
+    for (
+      let element = selectedElement();
+      element;
+      element = element.parentElement
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/components/quote-button.js.es6]: Do we need to climb the whole DOM tree up to the "root"? Could we just instead bail out as soon as we're out of the body of a post?

File: app/assets/javascripts/discourse/components/quote-button.js.es6 [hunk 2]
```diff
@@ -48,8 +48,28 @@ export default Component.extend({
...
+    ) {
+      if (element.tagName === "ASIDE" && element.classList.contains("quote")) {
+        opts = {
+          username:
+            element.dataset.username ||
+            element
+              .querySelector(".title")
+              .textContent.trim()
+              .replace(/:$/, ""),
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/components/quote-button.js.es6]: When do we have `$` in the username of a quote?

File: app/assets/javascripts/pretty-text/engines/discourse-markdown/quotes.js.es6 [hunk 3]
```diff
@@ -73,6 +73,10 @@ const rule = {
       token.attrs.push(["class", "quote no-group"]);
     }
 
+    if (username) {
+      token.attrs.push(["data-username", username]);
```
Comment (ZogStriP) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/quotes.js.es6]: Does this mean we need to rebake all the posts with a quote?

File: app/assets/javascripts/discourse/components/quote-button.js.es6 [hunk 4]
```diff
@@ -165,8 +185,8 @@ export default Component.extend({
   },
 
   click() {
-    const { postId, buffer } = this.quoteState;
-    this.attrs.selectText(postId, buffer).then(() => this._hideButton());
+    const { postId, buffer, opts } = this.quoteState;
+    this.attrs.selectText(postId, buffer, opts).then(() => this._hideButton());
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/components/quote-button.js.es6]: Think this should work

```suggestion
    this.attrs.selectText(postId, buffer, opts).then(this._hideButton);
```

**Pull request merged**